diff --git a/assets/js/autocomplete_deluxe.js b/assets/js/autocomplete_deluxe.js
index 893fbd9..05c8ca6 100755
--- a/assets/js/autocomplete_deluxe.js
+++ b/assets/js/autocomplete_deluxe.js
@@ -135,7 +135,7 @@
     this.uri = settings.uri;
     this.multiple = settings.multiple;
     this.required = settings.required;
-    this.limit = settings.limit;
+    this.match_limit = settings.match_limit;
     this.synonyms = typeof settings.use_synonyms == 'undefined' ? false : settings.use_synonyms;
     this.not_found_message = typeof settings.use_synonyms == 'undefined' ? Drupal.t("The entity '@term' will be added.") : settings.not_found_message;
     this.not_found_message_allow = typeof settings.not_found_message_allow == 'undefined' ? false : settings.not_found_message_allow;
diff --git a/src/Element/AutocompleteDeluxeElement.php b/src/Element/AutocompleteDeluxeElement.php
index 7232a16..2912091 100644
--- a/src/Element/AutocompleteDeluxeElement.php
+++ b/src/Element/AutocompleteDeluxeElement.php
@@ -96,7 +96,7 @@
       'input_id' => $html_id,
       'multiple' => $element['#multiple'],
       'required' => $element['#required'],
-      'limit' => isset($element['#limit']) ? $element['#limit'] : 10,
+      'match_limit' => isset($element['#match_limit']) ? $element['#match_limit'] : 10,
       'min_length' => isset($element['#min_length']) ? $element['#min_length'] : 0,
       'use_synonyms' => isset($element['#use_synonyms']) ? $element['#use_synonyms'] : 0,
       'delimiter' => isset($element['#delimiter']) ? $element['#delimiter'] : '',
diff --git a/src/Plugin/Field/FieldWidget/AutocompleteDeluxeWidget.php b/src/Plugin/Field/FieldWidget/AutocompleteDeluxeWidget.php
index 8685a2c..de310d6 100644
--- a/src/Plugin/Field/FieldWidget/AutocompleteDeluxeWidget.php
+++ b/src/Plugin/Field/FieldWidget/AutocompleteDeluxeWidget.php
@@ -108,7 +108,7 @@
       'autocomplete_route_name' => 'autocomplete_deluxe.autocomplete',
       'size' => 60,
       'selection_handler' => 'default',
-      'limit' => 10,
+      'match_limit' => 10,
       'min_length' => 0,
       'delimiter' => '',
       'not_found_message_allow' => FALSE,
@@ -122,11 +122,11 @@
    * {@inheritdoc}
    */
   public function settingsForm(array $form, FormStateInterface $form_state) {
-    $element['limit'] = [
+    $element['match_limit'] = [
       '#type' => 'textfield',
       '#title' => $this->t('Limit of the output.'),
       '#description' => $this->t('If set to zero no limit will be used'),
-      '#default_value' => $this->getSetting('limit'),
+      '#default_value' => $this->getSetting('match_limit'),
       '#element_validate' => [[get_class($this), 'validateInteger']],
     ];
     $element['min_length'] = [
@@ -177,7 +177,7 @@
   public function settingsSummary() {
     $summary = [];
 
-    $summary[] = $this->t('Limit: @limit', ['@limit' => $this->getSetting('limit')]);
+    $summary[] = $this->t('Limit: @match_limit', ['@match_limit' => $this->getSetting('match_limit')]);
     $summary[] = $this->t('Min length: @min_length', ['@min_length' => $this->getSetting('min_length')]);
     $summary[] = $this->t('Delimiter: @delimiter', ['@delimiter' => $this->getSetting('delimiter')]);
     $summary[] = $this->t('Allow Not Found message: @not_found_message_allow', ['@not_found_message_allow' => $this->getSetting('not_found_message_allow') ? 'Yes' : 'No']);
@@ -203,7 +203,7 @@
 
     $allow_message = $settings['not_found_message_allow'] ?? FALSE;
     $not_found_message = $settings['not_found_message'] ?? "";
-
+    $selection_settings['match_limit'] = isset($settings['match_limit']) ? $settings['match_limit'] : 10;
 
     $element += [
       '#type' => 'autocomplete_deluxe',
@@ -212,7 +212,7 @@
       '#selection_handler' => $this->getFieldSetting('handler'),
       '#selection_settings' => $selection_settings,
       '#size' => 60,
-      '#limit' => $settings['limit'] ?? 10,
+      '#match_limit' => $settings['match_limit'] ?? 10,
       '#min_length' => $settings['min_length'] ?? 0,
       '#delimiter' => $settings['delimiter'] ?? '',
       '#not_found_message_allow' => $allow_message,
